consumer-a/src/index.ts

```diff
@@ -1,8 +1,18 @@
 import { get } from "superagent";
 import { Product } from "./types";
 
-export async function getProducts( baseUrl: string ): Promise<Product[]> {
+export type API = {
+    getProducts: () => Promise<Product[]>
+}
+
+export function api(baseUrl: string): API {
```

Comment:
feedback:

In general i am not a fan of adjusting the source code to satisfy testing requirement, I think this is not the way to go. At least in my opinion.

Reply:
You don't have to do this - I was thinking about adding a second endpoint `/products/:byId` to show a possibly more interesting usecase (with 201, 404, and https://http.cat/405) but I haven't had the time to start putting it in code. That was just the first refactoring for that step to be able to share the baseUrl and get that argument out of the way. I should propably have stashed it for a different commit, sorry

Reply:
> You don't have to do this - I was thinking about adding a second endpoint `/products/:byId` to show a possibly more interesting usecase (with 201, 404, and https://http.cat/405) but I haven't had the time to start putting it in code. That was just the first refactoring for that step to be able to share the baseUrl and get that argument out of the way. I should propably have stashed it for a different commit, sorry

yeah :) i have prepared it in cosnumer-c, but also not finished it ;)